src/BinSkim.Rules/StringToVersionMap.cs

```diff
@@ -11,4 +11,25 @@ namespace Microsoft.CodeAnalysis.IL.Rules
     public class StringToVersionMap : TypedPropertiesDictionary<Version>
     {
     }
+
+    public static class StringToVersionMapExtensions
+    {
+        internal static readonly Version s_maxVersion = new Version(int.MaxValue, int.MaxValue, int.MaxValue, int.MaxValue);
+        internal static readonly Version s_minVersion = new Version(0, 0, 0, 0);
+
+        public static Version GetVersionByKey(this StringToVersionMap stringToVersionMap, string key, bool returnMaxValueIfKeyDoesNotExist = false)
```

Comment:
>returnMaxValueIfKeyDoesNotExist [](http://example.com/codeflow?start=107&length=31)

In case the property is missing, this means that the user removed the property because that wasn't relevant. So, we should default to minValue. Otherwise, we would require the user to fix the version of the compiler.